File: fixtures/scale/lua/large/checks/check_emberstream.lua
local emberstream = require("ciphers.emberstream")
local emberstream_spec = require("specs.emberstream_spec")
local corpus_stream = require("corpus.corpus_stream")

local function check_emberstream()
  local spec = emberstream_spec.get()
  for _, text in ipairs(corpus_stream.corpus_stream()) do
    local sealed = emberstream.emberstream_encrypt(text, spec.key)
    local opened = emberstream.emberstream_decrypt(sealed, spec.key)
    if opened ~= text then return false end
  end
  return spec.category == "stream"
end

return { check_emberstream = check_emberstream }

File: fixtures/scale/typescript/large/src/checks/CheckEmberstream.ts
import { Emberstream } from '../ciphers/Emberstream';
import { EmberstreamSpec } from '../specs/EmberstreamSpec';
import { CorpusStream } from '../corpus/CorpusStream';

export class CheckEmberstream {
  static run(): boolean {
    const spec = EmberstreamSpec.get();
    for (const text of CorpusStream.texts()) {
      const sealed = Emberstream.encrypt(text, spec.key);
      const opened = Emberstream.decrypt(sealed, spec.key);
      if (opened !== text) return false;
    }
    return spec.category === 'stream';
  }
}
